### ISSUE TITLE
SmallMoleculeSetProposalEngine attempting to build HOH residues?

### ISSUE THREAD

**Issue by jchodera** [choderalab/perses#144]
Just saw this one:
```
Testing expanded ensemble sampler with T4LysozymeInhibitorsTestSystem 'explicit'
--------------------------------------------------------------------------------
Expanded Ensemble sampler iteration        0
................................................................................
MCMC sampler iteration 5
Taking 5 steps of Langevin dynamics...
Final energy is    -8878.808 kT
................................................................................
Updating chemical state with ncmc-geometry-ncmc scheme...
Proposing new topology...
Proposed transformation: c1ccc(cc1)CCCS => CCI
Generating new system...
Performing NCMC annihilation
LOG: Creating alchemically modified system...
LOG: Elapsed time 0.220 s.
LOG: Creating alchemically modified intermediate...
LOG: Elapsed time 0.050 s.
LOG: NCMC logP     -121.9 | initial_total_energy   -17227.0 kT | final_total_energy   -16923.1 kT.
Geometry engine proposal...
Traceback (most recent call last):
  File "perses/tests/test_samplers.py", line 114, in <module>
    t()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 772, in run
    self.update()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 752, in update
    self.update_state()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 694, in update_state
    geometry_logp_reverse = self.geometry_engine.logp_reverse(topology_proposal, geometry_new_positions, geometry_old_positions, self.sampler.thermodynamic_state.beta)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 119, in logp_reverse
    logp_proposal, _ = self._logp_propose(top_proposal, old_coordinates, beta, new_positions=new_coordinates, direction='reverse')
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 165, in _logp_propose
    atom_proposal_order, logp_choice = proposal_order_tool.determine_proposal_order(direction='reverse')
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 875, in determine_proposal_order
    chosen_torsion, logp_choice = self._choose_torsion(atoms_with_positions, atom)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 923, in _choose_torsion
    raise Exception("No eligible torsions found for placing atom %s." % str(atom_for_proposal))
Exception: No eligible torsions found for placing atom <Atom H2 [1941]; In HOH 640>.
```
I'm confused by every part of this...

**Comment by jchodera** [comment]
Hmmm...
```
Testing expanded ensemble sampler with T4LysozymeInhibitorsTestSystem 'explicit'
--------------------------------------------------------------------------------
Expanded Ensemble sampler iteration        0
................................................................................
MCMC sampler iteration 5
Taking 5 steps of Langevin dynamics...
Final energy is    -8890.180 kT
................................................................................
Updating chemical state with ncmc-geometry-ncmc scheme...
Proposing new topology...
Proposed transformation: c1ccc(cc1)CCCS => CCCSSCCC
Generating new system...
Performing NCMC annihilation
LOG: Creating alchemically modified system...
LOG: Elapsed time 0.529 s.
LOG: Creating alchemically modified intermediate...
LOG: Elapsed time 0.051 s.
LOG: NCMC logP     -152.8 | initial_total_energy   -17305.8 kT | final_total_energy   -16924.7 kT.
Geometry engine proposal...
Traceback (most recent call last):
  File "perses/tests/test_samplers.py", line 114, in <module>
    t()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 772, in run
    self.update()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 752, in update
    self.update_state()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 694, in update_state
    geometry_logp_reverse = self.geometry_engine.logp_reverse(topology_proposal, geometry_new_positions, geometry_old_positions, self.sampler.thermodynamic_state.beta)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 119, in logp_reverse
    logp_proposal, _ = self._logp_propose(top_proposal, old_coordinates, beta, new_positions=new_coordinates, direction='reverse')
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 165, in _logp_propose
    atom_proposal_order, logp_choice = proposal_order_tool.determine_proposal_order(direction='reverse')
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 875, in determine_proposal_order
    chosen_torsion, logp_choice = self._choose_torsion(atoms_with_positions, atom)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 923, in _choose_torsion
    raise Exception("No eligible torsions found for placing atom %s." % str(atom_for_proposal))
Exception: No eligible torsions found for placing atom <Atom H2 [1941]; In HOH 640>.
```

**Comment by jchodera** [comment]
I wonder if this is related to #140

**Comment by jchodera** [comment]
This may have been due to me incompletely merging a few different updates.  Closing for now, but will update if it happens again!

**Comment by jchodera** [comment]
Whoops, I guess not:
```
Testing expanded ensemble sampler with T4LysozymeInhibitorsTestSystem 'explicit'
--------------------------------------------------------------------------------
Expanded Ensemble sampler iteration        0
................................................................................
MCMC sampler iteration 5
Taking 5 steps of Langevin dynamics...
Final energy is    -8854.235 kT
................................................................................
Updating chemical state with ncmc-geometry-ncmc scheme...
Proposing new topology...
Proposed transformation: c1ccc(cc1)CCCS => c1cc(ccc1I)I
Generating new system...
Performing NCMC annihilation
LOG: Creating alchemically modified system...
LOG: Elapsed time 0.221 s.
LOG: Creating alchemically modified intermediate...
LOG: Elapsed time 0.049 s.
LOG: NCMC logP      -88.3 | initial_total_energy   -16860.7 kT | final_total_energy   -16640.4 kT.
Geometry engine proposal...
Traceback (most recent call last):
  File "perses/tests/test_samplers.py", line 114, in <module>
    t()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 772, in run
    self.update()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 752, in update
    self.update_state()
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/samplers/samplers.py", line 694, in update_state
    geometry_logp_reverse = self.geometry_engine.logp_reverse(topology_proposal, geometry_new_positions, geometry_old_positions, self.sampler.thermodynamic_state.beta)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 119, in logp_reverse
    logp_proposal, _ = self._logp_propose(top_proposal, old_coordinates, beta, new_positions=new_coordinates, direction='reverse')
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 165, in _logp_propose
    atom_proposal_order, logp_choice = proposal_order_tool.determine_proposal_order(direction='reverse')
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 885, in determine_proposal_order
    chosen_torsion, logp_choice = self._choose_torsion(atoms_with_positions, atom)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/perses-0.1-py2.7.egg/perses/rjmc/geometry.py", line 933, in _choose_torsion
    raise Exception("No eligible torsions found for placing atom %s." % str(atom_for_proposal))
Exception: No eligible torsions found for placing atom <Atom H2 [1947]; In HOH 642>.
```

**Comment by pgrinaway** [comment]
This has been resolved in #150
